internal/distro/rhel84/distro_v2.go

```diff
@@ -0,0 +1,745 @@
+	} else {
+		p.AddStage(osbuild.NewLocaleStage(&osbuild.LocaleStageOptions{Language: "en_US"}))
+	}
+	p.AddStage(osbuild.NewOSTreeInitStage(&osbuild.OSTreeInitStageOptions{Path: "/var/www/html/repo"}))
+
+	p.AddStage(osbuild.NewOSTreePullStage(
+		&osbuild.OSTreePullStageOptions{Repo: "/var/www/html/repo"},
+		t.ostreePullStageInputs("org.osbuild.pipeline", "name:ostree-commit", t.OSTreeRef()),
+	))
+	return p
+}
+
+func (t *imageTypeS2) containerPipeline() *osbuild.Pipeline {
+	p := new(osbuild.Pipeline)
+	// NOTE(akoutsou) 1to2t: final pipeline should always be named "assembler"
+	p.Name = "assembler"
+	p.Build = "name:build"
+	options := &osbuild.OCIArchiveStageOptions{
+		Architecture: t.arch.Name(),
+		Filename:     t.Filename(),
+		Config: &osbuild.OCIArchiveConfig{
+			Cmd:          []string{"httpd", "-D", "FOREGROUND"},
+			ExposedPorts: []string{"80"},
+		},
+	}
+	baseInput := new(osbuild.OCIArchiveStageInput)
+	baseInput.Type = "org.osbuild.tree"
+	baseInput.Origin = "org.osbuild.pipeline"
+	baseInput.References = []string{"name:container-tree"}
+	inputs := &osbuild.OCIArchiveStageInputs{Base: baseInput}
+	p.AddStage(osbuild.NewOCIArchiveStage(options, inputs))
+	return p
+}
+
+func (t *imageTypeS2) anacondaTreePipeline(repos []rpmmd.RepoConfig, packages []rpmmd.PackageSpec, options distro.ImageOptions, kernelVer string) *osbuild.Pipeline {
+	p := new(osbuild.Pipeline)
+	p.Name = "anaconda-tree"
+	p.Build = "name:build"
+	p.AddStage(osbuild.NewRPMStage(t.rpmStageOptions(repos), t.rpmStageInputs(packages)))
+	p.AddStage(osbuild.NewOSTreeInitStage(&osbuild.OSTreeInitStageOptions{Path: "/ostree/repo"}))
+	p.AddStage(osbuild.NewOSTreePullStage(
+		&osbuild.OSTreePullStageOptions{Repo: "/ostree/repo"},
+		t.ostreePullStageInputs("org.osbuild.source", options.OSTree.Parent, options.OSTree.Ref),
+	))
+	p.AddStage(osbuild.NewBuildstampStage(t.buildStampStageOptions()))
+	p.AddStage(osbuild.NewLocaleStage(&osbuild.LocaleStageOptions{Language: "en_US.UTF-8"}))
+
+	rootPassword := ""
+	rootUser := osbuild.UsersStageOptionsUser{
+		Password: &rootPassword,
+	}
+
+	installUID := 0
+	installGID := 0
+	installHome := "/root"
+	installShell := "/usr/libexec/anaconda/run-anaconda"
+	installPassword := ""
+	installUser := osbuild.UsersStageOptionsUser{
+		UID:      &installUID,
+		GID:      &installGID,
+		Home:     &installHome,
+		Shell:    &installShell,
+		Password: &installPassword,
+	}
+	usersStageOptions := &osbuild.UsersStageOptions{
+		Users: map[string]osbuild.UsersStageOptionsUser{
+			"root":    rootUser,
+			"install": installUser,
+		},
+	}
+
+	p.AddStage(osbuild.NewUsersStage(usersStageOptions))
+	p.AddStage(osbuild.NewAnacondaStage(t.anacondaStageOptions()))
+	p.AddStage(osbuild.NewLoraxScriptStage(t.loraxScriptStageOptions()))
+	p.AddStage(osbuild.NewDracutStage(t.dracutStageOptions(kernelVer)))
+	p.AddStage(osbuild.NewKickstartStage(t.kickstartStageOptions(options.OSTree.URL)))
```

Comment:
Since the commit is embedded into the installer, this should not be `options.OSTree.URL` but `file:///ostree/repo`